src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/Providers/AzureFilesPsBackupProvider.cs

```diff
@@ -62,70 +62,313 @@ public class AzureFilesPsBackupProvider : IPsBackupProvider
+                    }
+                    extendedInfo.PolicyState = serviceClientExtendedInfo.PolicyState.ToString();
+                    extendedInfo.RecoveryPointCount =
+                        (int)(serviceClientExtendedInfo.RecoveryPointCount.HasValue ?
+                            serviceClientExtendedInfo.RecoveryPointCount : 0);
+                    ((AzureFileShareItem)itemModel).ExtendedInfo = extendedInfo;
+                });
+
+            // 3. Filter by item's Protection Status
+            if (protectionStatus != 0)
+            {
+                itemModels = itemModels.Where(itemModel =>
+                {
+                    return ((AzureFileShareItem)itemModel).ProtectionStatus == protectionStatus;
+                }).ToList();
+            }
+
+            // 4. Filter by item's Protection State
+            if (status != 0)
+            {
+                itemModels = itemModels.Where(itemModel =>
+                {
+                    return ((AzureFileShareItem)itemModel).ProtectionState == status;
+                }).ToList();
+            }
+
+            // 5. Filter by workload type
+            if (workloadType != 0)
+            {
+                itemModels = itemModels.Where(itemModel =>
+                {
+                    return itemModel.WorkloadType == workloadType;
+                }).ToList();
+            }
+
+            return itemModels;
+        }
+
+        private RestAzureNS.AzureOperationResponse EnableOrModifyProtection()
+        {
+            string vaultName = (string)ProviderData[VaultParams.VaultName];
+            string vaultResourceGroupName = (string)ProviderData[VaultParams.ResourceGroupName];
+            string azureFileShareName = ProviderData.ContainsKey(ItemParams.ItemName) ?
+                (string)ProviderData[ItemParams.ItemName] : null;
+            string storageAccountName = ProviderData.ContainsKey(ItemParams.StorageAccountName) ?
+                (string)ProviderData[ItemParams.StorageAccountName] : null;
+            string parameterSetName = ProviderData.ContainsKey(ItemParams.ParameterSetName) ?
+                (string)ProviderData[ItemParams.ParameterSetName] : null;
+
+            PolicyBase policy = ProviderData.ContainsKey(ItemParams.Policy) ?
+                (PolicyBase)ProviderData[ItemParams.Policy] : null;
+            bool deleteBackupData = ProviderData.ContainsKey(ItemParams.DeleteBackupData) ?
+                (bool)ProviderData[ItemParams.DeleteBackupData] : false;
+
+            ItemBase itemBase = (ItemBase)ProviderData[ItemParams.Item];
+
+            AzureFileShareItem item = (AzureFileShareItem)ProviderData[ItemParams.Item];
+
+            string containerUri = "";
+            string protectedItemUri = "";
+            string sourceResourceId = null;
+            AzureFileshareProtectedItem properties = new AzureFileshareProtectedItem();
+
+            if (policy != null)
```

Comment:
Please don't depend on policy being null for this check. Expose a boolean parameter bool disableWithRetainData and use this for this check. Also, please add some comments so that the flow will be a bit more readable.